sycl/test-e2e/AddressSanitizer/out-of-bounds/USM/matrix_multiply.cpp

```diff
@@ -0,0 +1,105 @@
+// REQUIRES: linux
+// RUN: %{build} %device_sanitizer_flags -DMALLOC_DEVICE -O1 -g -o %t
+// RUN: %{run} %t &> %t.txt ; FileCheck --check-prefixes CHECK,CHECK-DEVICE --input-file %t.txt %s
+// RUN: %{build} %device_sanitizer_flags -DMALLOC_DEVICE -O2 -g -o %t
+// RUN: %{run} %t &> %t.txt ; FileCheck --check-prefixes CHECK,CHECK-DEVICE --input-file %t.txt %s
+// RUN: %{build} %device_sanitizer_flags -DMALLOC_HOST -O2 -g -o %t
+// RUN: %{run} %t &> %t.txt ; FileCheck --check-prefixes CHECK,CHECK-HOST --input-file %t.txt %s
+// RUN: %{build} %device_sanitizer_flags -DMALLOC_SHARED -O2 -g -o %t
+// RUN: %{run} %t &> %t.txt ; FileCheck --check-prefixes CHECK,CHECK-SHARED --input-file %t.txt %s
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 16;
+  constexpr std::size_t M = 16;
+  constexpr std::size_t K = N / 4;
+#if defined(MALLOC_HOST)
+  auto matrixA = (int(*)[N])sycl::malloc_host<int>(N * M, Q);
+  auto matrixB = (int(*)[N])sycl::malloc_host<int>(N * M, Q);
+  auto matrixC = (int(*)[N])sycl::malloc_host<int>(N * M, Q);
+#elif defined(MALLOC_SHARED)
+  auto matrixA = (int(*)[N])sycl::malloc_shared<int>(N * M, Q);
+  auto matrixB = (int(*)[N])sycl::malloc_shared<int>(N * M, Q);
+  auto matrixC = (int(*)[N])sycl::malloc_shared<int>(N * M, Q);
+#elif defined(MALLOC_DEVICE)
+  auto matrixA = (int(*)[N])sycl::malloc_device<int>(N * M, Q);
+  auto matrixB = (int(*)[N])sycl::malloc_device<int>(N * M, Q);
+  auto matrixC = (int(*)[N])sycl::malloc_device<int>(N * M, Q);
+#elif defined(MALLOC_SYSTEM)
```

Comment:
these two section can be merged within  "defined(MALLOC_DEVICE) || defined(MALLOC_SYSTEM)"

Reply:
the MALLOC_SYSTEM is wrong code, I changed it to correct one.